### PR TITLE
Add support for transporting product ID in events

### PR DESCRIPTION
**Pull request by ncooke3**

### CODE REVIEW

File: GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019 Google
+ * Copyright 2022 Google LLC
```
Comment (paulb777) [inline GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.h:2]: Doesn't have to change, but if it does, it should be to 2023

Comment (ncooke3) [reply GoogleDataTransport/GDTCCTLibrary/Protogen/nanopb/cct.nanopb.h]: Thanks – I reverted the change. When running the proto generation script, it updates all the copyrights with copyright notice in the script. If it's alright with you, I might just let it update in the future so I need to revert all of related changes.

File: GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h [hunk 2]
```diff
@@ -74,6 +75,9 @@ typedef NS_ENUM(NSInteger, GDTCOREventQoS) {
 /** Bytes that can be used by an uploader later on. */
 @property(nullable, nonatomic) NSData *customBytes;
 
+/** The product data that the event is associated with, if any.  */
+@property(nullable, readonly, nonatomic) GDTCORProductData *productData;
```
Comment (paulb777) [inline GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h:79]: What's the reason for the productData class versus adding a productID property to GDTCorEvent?

Comment (ncooke3) [reply GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h:79]: Discussed offline.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**